Packs/ApiModules/Scripts/EWSApiModule/EWSApiModule.py

```diff
@@ -0,0 +1,718 @@
+            raise original_exc  # pylint: disable=E0702
+
+        if target_mailbox == self.account_email:
+            demisto.setIntegrationContext(new_context)
+
+        return account
+
+    def get_items_from_mailbox(self, account: Optional[Union[Account, str]], item_ids):
+        """
+        Request specific items from a mailbox associated with an account
+
+        :param account: EWS account or target_mailbox associated with that account
+        :param item_ids: item_ids of the requested items
+
+        :return: list of exchangelib Items
+        """
+        # allow user to pass target_mailbox as account
+        if not isinstance(account, Account):
+            account = self.get_account(account) if isinstance(account, str) else self.get_account(self.account_email)
+
+        if not isinstance(item_ids, list):
+            item_ids = [item_ids]
+
+        items = [Item(id=x) for x in item_ids]
+        result = list(account.fetch(ids=items))
+        result = [x for x in result if not (isinstance(x, ErrorItemNotFound | ErrorInvalidIdMalformed))]
+        if len(result) != len(item_ids):
+            raise Exception('One or more items were not found/malformed. Check the input item ids')
+        return result
+
+    def get_item_from_mailbox(self, account: Optional[Union[Account, str]], item_id):
+        """
+        Request a single item from a mailbox associated with an account
+
+        :param account: EWS account or target_mailbox associated with that account
+        :param item_id: item_id of the requested item
+
+        :return: exchangelib Item
+        """
+        result = self.get_items_from_mailbox(account, [item_id])
+        if len(result) == 0:
+            raise Exception(f'ItemId {str(item_id)} not found')
+        return result[0]
+
+    def get_attachments_for_item(self, item_id, account: Optional[Union[Account, str]], attachment_ids: list = []):
+        """
+        Request attachments for an item
+
+        :param item_id: item_id of the item to retrieve attachments from
+        :param account: EWS account or target_mailbox associated with that account
+        :param (Optional) attachment_ids: attachment_ids to retrieve, empty list will get all available attachments
+
+        :return: list of exchangelib Item.attachments
+        """
+        item = self.get_item_from_mailbox(account, item_id)
+        if not item:
+            raise DemistoException(f'Message item not found: {item_id}')
+
+        attachments = []
+        for attachment in item.attachments or []:
+            if attachment is None:
+                continue
+
+            if not attachment_ids or attachment.attachment_id.id in attachment_ids:
+                attachments.append(attachment)
+
+        if attachment_ids and len(attachments) < len(attachment_ids):
+            raise DemistoException(f'Some attachment id was not found for message: {str(attachment_ids)}')
```

Comment:
Try to add a more informative log how much attachments were not found and what are the values? 
Also if some the please use were.